lib/internal/modules/esm/default_resolve.js

```diff
@@ -24,6 +25,10 @@ function search(target, base) {
     // We cannot search without a base.
     throw new ERR_MISSING_MODULE(target);
   }
+  if ((target[0] === '.' || target[0] === '/') &&
```

Comment:
This check should be `target[0] === '.' && (target[1] === '/' || target[1] === '.' && target[2] === '.') || target[0] === '/'` to ensure support for filenames starting with `.`. Alternatively just the regex `/^\.{0,2}\//.test(specifier)`.

I understand this is an initial implementation, but note that this implementation still supports "main"s and URLs getting default extensions. Happy to do a deeper review in due course.

Reply:
maybe `String.prototype.startsWith` can be used?

Reply:
If we do use a method, i'd prefer the non-taintable approaches used elsewhere for robustness